Plugin/Framework/Search/Request/GraphQlHandler.php

```diff
@@ -116,6 +119,10 @@ protected function preFetchOps(array $requestData)
     // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter
     public function parseLimit(array $requestData)
     {
+        if ($this->pageSize != -1) {
```

Comment:
Is this even possible? If so, shouldn't you be throwing this exception in the constructor when an invalid constructor arg is specified?

Reply:
@mridang idea is that if the page size is set to -1 in di.xml, as it is by default, then the default page size of the request will be used, otherwise the merchant can override these values.

Reply:
Why can't the default value be in the DI configuration? Also, why can't this configuration value and default-if-null (if needed) be done in the constructor?